recipes/fpng/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import get, save, load
+from conan.tools.build import check_min_cppstd
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class FpngConan(ConanFile):
+    name = "fpng"
+    description = "Super fast C++ .PNG writer/reader"
+    license = "unlicense",
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/richgel999/fpng"
+    topics = ("png", "writer", "reader")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_sse": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_sse": True,
+    }
+    exports_sources = ["CMakeLists.txt"]
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["FPNG_SRC_DIR"] = self.source_folder.replace("\\", "/")
```

Comment:
Not worth changing it here, but I generally find it easier to
```py
def export_sources(self):
    copy(self, "CMakeLists.txt", self.recipe_folder, os.path.join(self.export_sources_folder, "src"))
```
That way all of the CMakeLists.txt and source paths work without any prefixing.

The current approach is not wrong either, though, of course.

Reply:
@valgur 
Sorry for my late response.

I tried your comment, but I failed.
Are there any misunderstandings in my modification?

Error message is following.

```
fpng/1.0.6: Running CMake.configure()
fpng/1.0.6: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/toge/.conan2/p/b/fpng8db6eab8b7baf/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/toge/.conan2/p/b/fpng8db6eab8b7baf/b/src"
CMake Warning:
  Ignoring extra path from command line:

   "/home/toge/.conan2/p/b/fpng8db6eab8b7baf/b/src"


CMake Error: The source directory "/home/toge/.conan2/p/b/fpng8db6eab8b7baf/b/src" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
```

modified conanfile.py
<details>
```
from conan import ConanFile
from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
from conan.tools.files import get, save, load, copy, rm
from conan.tools.build import check_min_cppstd
import os

required_conan_version = ">=1.53.0"


class FpngConan(ConanFile):
    name = "fpng"
    description = "Super fast C++ .PNG writer/reader"
    license = "Unlicense",
    url = "https://github.com/conan-io/conan-center-index"
    homepage = "https://github.com/richgel999/fpng"
    topics = ("png", "writer", "reader")
    package_type = "library"
    settings = "os", "arch", "compiler", "build_type"
    options = {
        "shared": [True, False],
        "fPIC": [True, False],
        "with_sse": [True, False],
    }
    default_options = {
        "shared": False,
        "fPIC": True,
        "with_sse": True,
    }

    def export_sources(self):
        copy(self, "CMakeLists.txt", self.recipe_folder, os.path.join(self.export_sources_folder, "src"))

    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
            self.options.rm_safe("fPIC")

    def layout(self):
        cmake_layout(self, src_folder="src")

    def validate(self):
        check_min_cppstd(self, 11)

    def source(self):
        get(self, **self.conan_data["sources"][self.version], strip_root=True)
        # remove original CMakeLists.txt (which builds a test program only)
        rm(self, "CMakeLists.txt", self.source_folder)

    def generate(self):
        tc = CMakeToolchain(self)
        tc.variables["FPNG_WITH_SSE"] = self.options.with_sse
        tc.generate()

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def package(self):
        filename = os.path.join(self.source_folder, "src", "fpng.cpp")
        file_content = load(self, filename)
        license_start = "	This is free and unencumbered software"
        license_end = "*/"
        license_contents = file_content[file_content.find(license_start):file_content.find(license_end)]
        save(self, os.path.join(self.package_folder, "licenses", "LICENSE"), license_contents)
        cmake = CMake(self)
        cmake.install()

    def package_info(self):
        self.cpp_info.libs = ["fpng"]
        if self.settings.os in ("FreeBSD", "Linux"):
            self.cpp_info.system_libs = ["m"]
```
</details>